proteome-rest/src/main/java/org/uniprot/api/proteome/service/GeneCentricService.java

```diff
@@ -20,40 +17,27 @@
  * @date: 30 Apr 2019
  */
 @Service
-public class GeneCentricService {
-    private final GeneCentricSortClause solrSortClause;
-    private GeneCentricFacetConfig facetConfig;
-    private final BasicSearchService<CanonicalProtein, GeneCentricDocument> basicService;
-    private final DefaultSearchHandler defaultSearchHandler;
+public class GeneCentricService extends BasicSearchService<GeneCentricDocument, CanonicalProtein> {
+    private static final Supplier<DefaultSearchHandler> handlerSupplier =
```

Comment:
what benefit does the function provide?

Reply:
Readability. Other option was to call the constructor of DefaulSearchHandler with 3 params in super call which will make the statement long.